### PR TITLE
S28 2277/portal access

### PR DESCRIPTION
**Pull request by mazzopardi2**
**Before creating a pull request make sure that:**

- [X ] commit messages are meaningful and follow good commit message guidelines
- [ X] README and other documentation has been updated / added (if needed)
- [ X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/S28-2467 

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: bin/migration/migration_reports/failed_imports_logger.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                recording_id = failed_import.get('recording_id', 'N/A')
+                details = failed_import.get('details')
+
+                if details is not None:
+                    file.write(f"| {str(table_name).ljust(22)} | {str(table_id).ljust(36)} | {str(case_id).ljust(36)} | {str(recording_id).ljust(36)} | {str(details)} \n")
+
+                
+            # if (table_name, failed_id) not in self.existing_entries_cache:
+            #     self.existing_entries_cache.add((table_name.strip(), failed_id.strip())) 
+                # file.write(f"Table: {table_name}, ID: {failed_id}, Details: {details}\n")
```
Comment (zc-hmcts) [inline bin/migration/migration_reports/failed_imports_logger.py:43]: don't need to keep commented out blocks

File: bin/migration/tables/courtregions.py [hunk 2]
```diff
@@ -19,47 +19,51 @@ def migrate_data(self,destination_cursor):
             {"name": "Leeds Crown Court", "region": "Yorkshire and The Humber"},
             {"name": "Leeds", "region": "Yorkshire and The Humber"},
             {"name": "Durham Crown Court", "region": "North East (England)"},
-            {"name": "Liverpool Crown Court", "region": "North West (England)"},
-            {"name": "Nottingham Crown Court", "region": "East Midlands (England)"},
+            {"name": "Liverpool Crown Court","region": "North West (England)"},
```
Comment (zc-hmcts) [inline bin/migration/tables/courtregions.py:22]:
```suggestion
            {"name": "Liverpool Crown Court", "region": "North West (England)"},
```

File: bin/migration/tables/courtregions.py [hunk 3]
```diff
@@ -19,47 +19,51 @@ def migrate_data(self,destination_cursor):
...
+            {"name": "Nottingham Crown Court","region": "East Midlands (England)"},
```
Comment (zc-hmcts) [inline bin/migration/tables/courtregions.py:23]:
```suggestion
            {"name": "Nottingham Crown Court", "region": "East Midlands (England)"},
```

File: bin/migration/tables/appaccess.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def migrate_data(self, destination_cursor, source_data):
                 continue
 
             court_id = user[2]
-            active = True if user[3].lower() == "active" else False
+            active = True if user[3] and user[3].lower() == "active" else False
```
Comment (LaurenceRawlings) [inline bin/migration/tables/appaccess.py:41]:
```suggestion
            active = user[3] and user[3].lower() == "active"
```

**Review by LaurenceRawlings — APPROVED** [review]

**Pull request merged**